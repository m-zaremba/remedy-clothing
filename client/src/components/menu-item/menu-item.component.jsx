import React from 'react';
import { withRouter } from 'react-router-dom';
import PropTypes from 'prop-types';
import {
  MenuItemContainer,
  MenuItemTitle,
  MenuItemSubtitle,
  MenuItemContentContainer,
  BackgroundImage,
} from './menu-item.styles';

export const MenuItem = ({ title, imageUrl, size, history, linkUrl, match }) => {
  return (
    <MenuItemContainer
      size={size}
      role="button"
      tabIndex={0}
      onClick={() => history.push(`${match.url}${linkUrl}`)}
      onKeyDown={() => history.push(`${match.url}${linkUrl}`)}
    >
      <BackgroundImage
        style={{
          backgroundImage: `url(${imageUrl})`,
        }}
      />
      <MenuItemContentContainer>
        <MenuItemTitle>{title}</MenuItemTitle>
        <MenuItemSubtitle>SHOP NOW</MenuItemSubtitle>
      </MenuItemContentContainer>
    </MenuItemContainer>
  );
};

MenuItem.defaultProps = {
  size: null,
  linkUrl: null,
};

MenuItem.propTypes = {
  title: PropTypes.string.isRequired,
  imageUrl: PropTypes.string.isRequired,
  size: PropTypes.string,
  history: PropTypes.objectOf(PropTypes.shape).isRequired,
  linkUrl: PropTypes.string,
  match: PropTypes.objectOf(PropTypes.shape).isRequired,
};

export default withRouter(MenuItem);
