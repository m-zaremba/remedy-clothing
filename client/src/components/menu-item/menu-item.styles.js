import styled, { css } from 'styled-components';

const largeStyle = css`
  height: 350px;
`;

const regularStyle = css`
  height: 240px;
`;

const getMenuItemButtonStyles = (props) => {
  return props.size ? largeStyle : regularStyle;
};

export const BackgroundImage = styled.div`
  width: 100%;
  height: 100%;
  background-position: center;
  background-size: cover;
`;

export const MenuItemContentContainer = styled.div`
  height: 90px;
  padding: 0 25px;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  border: 1px solid black;
  background-color: rgb(255, 255, 255);
  opacity: 0.7;
  position: absolute;
`;

export const MenuItemContainer = styled.div`
  min-width: 30%;
  flex: 1 1 auto;
  display: flex;
  align-items: center;
  justify-content: center;
  border: 1px solid black;
  margin: 0 7.5px 15px;
  overflow: hidden;
  ${getMenuItemButtonStyles}

  &:first-child {
    margin-right: 7.5px;
  }

  &:last-child {
    margin-left: 7.5px;
  }

  &:hover {
    cursor: pointer;

    ${BackgroundImage} {
      transform: scale(1.1);
      transition: transform 6s cubic-bezier(0.25, 0.45, 0.45, 0.95);
    }

    ${MenuItemContentContainer} {
      opacity: 0.9;
    }
  }

  @media screen and (max-width: 800px) {
    height: 240px;
  }
`;

export const MenuItemTitle = styled.h1`
  font-weight: bold;
  margin-bottom: 6px;
  font-size: 22px;
  color: #4a4a4a;
  margin: 0;
  text-transform: uppercase;
`;

export const MenuItemSubtitle = styled.p`
  font-weight: lighter;
  font-size: 16px;
  margin: 0;
`;
