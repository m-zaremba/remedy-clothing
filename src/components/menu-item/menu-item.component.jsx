import React from 'react';
import './menu-item.styles.scss';
import { withRouter } from 'react-router-dom';
import PropTypes from 'prop-types';

const MenuItem = ({ title, imageUrl, size, history, linkUrl, match }) => {
  return (
    <div
      role="button"
      tabIndex={0}
      className={size ? `menu-item ${size}` : 'menu-item'}
      onClick={() => history.push(`${match.url}${linkUrl}`)}
      onKeyDown={() => history.push(`${match.url}${linkUrl}`)}
    >
      <div
        style={{
          backgroundImage: `url(${imageUrl})`,
        }}
        className="background-image"
      />
      <div className="content">
        <h1 className="title">{title}</h1>
        <p className="subtitle">SHOP NOW</p>
      </div>
    </div>
  );
};

MenuItem.defaultProps = {
  size: null,
  linkUrl: null,
};

MenuItem.propTypes = {
  title: PropTypes.string.isRequired,
  imageUrl: PropTypes.string.isRequired,
  size: PropTypes.string,
  history: PropTypes.objectOf(PropTypes.shape).isRequired,
  linkUrl: PropTypes.string,
  match: PropTypes.objectOf(PropTypes.shape).isRequired,
};

export default withRouter(MenuItem);
