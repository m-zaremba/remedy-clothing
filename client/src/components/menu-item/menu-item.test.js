import React from 'react';
import { shallow } from 'enzyme';

import { MenuItem } from './menu-item.component';

describe('testing MenuItem component', () => {
  let wrapper;
  let mockMatch;
  let mockHistory;
  const linkUrl = '/hats';
  const size = 'large';
  const imageUrl = 'www.testImageURL.com';

  beforeEach(() => {
    mockMatch = {
      url: '/shop',
    };

    mockHistory = {
      push: jest.fn(),
    };

    const mockProps = {
      match: mockMatch,
      history: mockHistory,
      linkUrl,
      size,
      title: 'hats',
      imageUrl,
    };

    wrapper = shallow(<MenuItem {...mockProps} />);
  });

  it('should render MenuItem component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call history.push with the right string when MenuItemContainer is clicked', () => {
    wrapper.find('MenuItemContainer').simulate('click');

    expect(mockHistory.push).toHaveBeenCalledWith(`${mockMatch.url}${linkUrl}`);
  });

  it('should pass size to MenuItemContainer as the prop size', () => {
    expect(wrapper.find('MenuItemContainer').prop('size')).toBe(size);
  });
});
