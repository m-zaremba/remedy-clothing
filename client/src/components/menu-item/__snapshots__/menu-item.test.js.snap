// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing MenuItem component should render MenuItem component 1`] = `
<MenuItemContainer
  onClick={[Function]}
  onKeyDown={[Function]}
  role="button"
  size="large"
  tabIndex={0}
>
  <styled.div
    style={
      Object {
        "backgroundImage": "url(www.testImageURL.com)",
      }
    }
  />
  <styled.div>
    <styled.h1>
      hats
    </styled.h1>
    <styled.p>
      SHOP NOW
    </styled.p>
  </styled.div>
</MenuItemContainer>
`;
